File: leetcode/20_Valid_Parentheses.py
strin="([]){}"
def validator(s):
    for i in range(0, len(s)):
        if(len(s) ==0):
            return True
        if(len(s)%2!=0):
            return False
        if(s[i] ==")"):
            if(s[i-1] == "("):
                return validator(s[:i-1]+s[i+1:])
            else:
                return False
        elif(s[i] == "}"):
            if(s[i-1] == "{"):
                return validator(s[:i-1]+s[i+1:])
            else:
                return False
        elif(s[i] == "]"):
            if(s[i-1] == "["):
                return validator(s[:i-1]+s[i+1:])
            else:
                return False


if(validator(strin)):
    print("valid parenthesis")
else:
    print("not valid parenthesis")
                        

File: leetcode/20_Valid_Parentheses.cpp
#include <iostream>
#include <string>
#include <stack>
#include<map>
using namespace std;
bool validator(string s){
    stack<char> st;
    map<char, char> map;
    map[']']='[';
    map['}']='{';
    map[')']='(';
    for (int i = 0; i < s.length(); i++)
    {
        if (s[i] == '(' || s[i] == '[' || s[i] == '{')
        {
            st.push(s[i]);
        }
        else if (s[i] == ')' || s[i] == ']' || s[i] == '}')
        {
            if ( map[s[i]]== st.top())
            {
                st.pop();
            }
            else{
                return false;
            }
        }
    }
    if (st.empty())
    {
        return true;
    }
    else
    {
       return false;
    }
}
int main()
{
    string s = "]";
    if(validator(s)){
        cout<<"valid";
    }
    else{
        cout<<"Invalid";
    }
    return 0;
}